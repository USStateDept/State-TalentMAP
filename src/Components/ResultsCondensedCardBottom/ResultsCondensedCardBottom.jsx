import React from 'react';
import PropTypes from 'prop-types';
import CondensedCardData from '../CondensedCardData';
import BidListButton from '../BidListButton';
import { POSITION_DETAILS, BID_RESULTS, FAVORITE_POSITIONS_ARRAY } from '../../Constants/PropTypes';
import Favorite from '../Favorite/Favorite';

const ResultsCondensedCardBottom = ({ position, toggleBid, bidList, toggleFavorite,
favorites }) => (
  <div className="condensed-card-bottom-container">
    <div className="usa-grid-full condensed-card-bottom">
      <CondensedCardData position={position} />
      <div className="usa-grid-full condensed-card-buttons-section">
        <Favorite
          hideText
          hasBorder
          refKey={position.id}
          onToggle={toggleFavorite}
          compareArray={favorites}
          useButtonClass
          useButtonClassSecondary
        />
        <BidListButton
          className="tm-button"
          id={position.id}
          compareArray={bidList}
          toggleBidPosition={toggleBid}
        />
      </div>
    </div>
  </div>
  );

ResultsCondensedCardBottom.propTypes = {
  position: POSITION_DETAILS.isRequired,
  toggleBid: PropTypes.func.isRequired,
  bidList: BID_RESULTS.isRequired,
  toggleFavorite: PropTypes.func.isRequired,
  favorites: FAVORITE_POSITIONS_ARRAY.isRequired,
};

ResultsCondensedCardBottom.defaultProps = {
  type: 'default',
};

export default ResultsCondensedCardBottom;
