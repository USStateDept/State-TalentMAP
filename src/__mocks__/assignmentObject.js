const assignmentObject = {
  start_date: '10-10-2016',
  end_date: '10-10-2018',
  estimated_end_date: '10-10-2018',
  position: {
    skill: 'CONSTRUCTION ENGINEERING (6218)',
    bureau: '(NEA) BUREAU NEAR EASTERN AFFAIRS',
    position_number: '58348065',
    title: 'PROJECT DIRECTOR',
    post: {
      location: {
        id: 4,
        country: 'United States',
        code: '500',
        city: 'Washington',
        state: 'DC',
      },
    },
  },
};

export default assignmentObject;
