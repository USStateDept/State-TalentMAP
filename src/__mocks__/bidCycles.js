export default [
  {
    id: 1,
    name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
    cycle_start_date: '2017-12-13T15:51:10.980554Z',
    cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
    cycle_end_date: '2018-06-13T15:51:10.980554Z',
    active: true,
  },
];
