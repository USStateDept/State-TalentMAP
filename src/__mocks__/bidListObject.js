const bidListObject = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 3,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 226,
        grade: '03',
        skill: 'POLITICAL AND ECONOMIC AFFAIRS (6050)',
        position_number: '10035561',
        title: 'POLITICAL/ECONOMIC OFFICER',
        create_date: '2015-08-05',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 199,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'approved',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: false,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 1,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 1,
        grade: '05',
        skill: 'OFFICE MANAGEMENT (9017)',
        position_number: '00003026',
        title: 'OMS (COM)',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 235,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'closed',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: true,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 2,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 84,
        grade: '04',
        skill: 'SECURITY (2501)',
        position_number: '56001600',
        title: 'SPECIAL AGENT',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 195,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'closed',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: false,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
  ],
};

export default bidListObject;
