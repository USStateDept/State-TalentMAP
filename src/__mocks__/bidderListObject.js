const bidderListObject = {
  count: 6,
  next: null,
  previous: null,
  results: [
    {
      id: 5,
      current_assignment: '(active) Kara Batisak at [58312265] PROJECT DIRECTOR (Guatemala, Guatemala)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 1,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Kara Batisak',
          draft: 6,
          submitted: 3,
          handshake_offered: 1,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'batisak',
        email: 'batisak@state.gov',
        first_name: 'Kara',
        last_name: 'Batisak',
      },
      language_qualifications: [],
    },
    {
      id: 3,
      current_assignment: '(active) John Doe at [10181012] INTERNATIONAL RELATIONS OFF (Singapore, Singapore)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'doej',
        email: 'doej@talentmap.us',
        first_name: 'John',
        last_name: 'Doe',
      },
      language_qualifications: [],
    },
    {
      id: 1,
      current_assignment: '(active) Guest McGuestson at [58312870] CONSTRUCTION MANAGER (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '03',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'guest',
        email: 'guest@state.gov',
        first_name: 'Guest',
        last_name: 'McGuestson',
      },
      language_qualifications: [],
    },
    {
      id: 6,
      current_assignment: '(active) Tarek Rehman at [56457009] ASST REGIONAL SECURITY OFCR (Adana, Turkey)',
      skills: [
        {
          id: 20,
          cone: null,
          code: '2501',
          description: 'SECURITY',
        },
      ],
      grade: '04',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 2,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Tarek Rehman',
          draft: 5,
          submitted: 2,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'rehmant',
        email: 'rehmant@state.gov',
        first_name: 'Tarek',
        last_name: 'Rehman',
      },
      language_qualifications: [],
    },
    {
      id: 2,
      current_assignment: '(active) Administrator TalentMAP at [58312869] PROJECT DIRECTOR (Nogales, Mexico)',
      skills: [
        {
          id: 63,
          cone: null,
          code: '6218',
          description: 'CONSTRUCTION ENGINEERING',
        },
      ],
      grade: '01',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [],
      user: {
        username: 'admin',
        email: 'admin@talentmap.us',
        first_name: 'Administrator',
        last_name: 'TalentMAP',
      },
      language_qualifications: [],
    },
    {
      id: 4,
      current_assignment: '(active) Jenny Townpost at [10026129] INTERNATIONAL RELATIONS OFF (Port Moresby, Papua New Guinea)',
      skills: [
        {
          id: 55,
          cone: null,
          code: '6080',
          description: 'INTERFUNCTIONAL OFFICER OPER',
        },
      ],
      grade: '02',
      is_cdo: false,
      primary_nationality: 'United States',
      secondary_nationality: null,
      bid_statistics: [
        {
          id: 4,
          bidcycle: 'Demo BidCycle 2018-01-10 15:52:20.583434',
          user: 'Jenny Townpost',
          draft: 3,
          submitted: 4,
          handshake_offered: 3,
          handshake_accepted: 0,
          handshake_declined: 0,
          in_panel: 0,
          approved: 0,
          declined: 0,
          closed: 0,
        },
      ],
      user: {
        username: 'townpostj',
        email: 'townpostj@state.gov',
        first_name: 'Jenny',
        last_name: 'Townpost',
      },
      language_qualifications: [],
    },
  ],
};

export default bidderListObject;
