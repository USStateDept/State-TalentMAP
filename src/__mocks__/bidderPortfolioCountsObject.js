const bidderPortfolioCountsObject = {
  all_clients: 100,
  bidding_clients: 50,
  in_panel_clients: 20,
  on_post_clients: 30,
  bidding_no_handshake: 10,
};

export default bidderPortfolioCountsObject;
