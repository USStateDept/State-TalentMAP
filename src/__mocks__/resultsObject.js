const resultsObject = {
  count: 2,
  results: [
    {
      id: 6,
      grade: '06',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
      organization: '(MONROVIA) MONROVIA LIBERIA',
      classifications: [],
      representation: '[10034001] OMS (DCM) (Monrovia, Liberia)',
      is_highlighted: false,
      position_number: '10034001',
      title: 'OMS (DCM)',
      is_overseas: true,
      create_date: '2006-09-20T00:00:00Z',
      update_date: '2017-06-08T00:00:00Z',
      effective_date: '2012-10-22T00:00:00Z',
      posted_date: '2012-10-22T00:00:00Z',
      status_code: null,
      status: null,
      description: {
        id: 90,
        last_editing_user: null,
        is_editable_by_user: false,
        date_created: '2018-03-08T18:00:33.241370Z',
        date_updated: '2018-03-13T15:55:00.018644Z',
        content: 'The Office Management Specialist (OMS) for the Deputy Chief of Mission (DCM) provides office management services in this Mission of six agencies, 387 American and Locally Employed Staff (LES) positions, and 279 guard force contractors.  The OMS manages Executive Office operations and positionstains the DCM’s calendar in support of overall Mission goals and objectives.  The OMS (DCM) drafts routine correspondence, memoranda and cables as appropriate and ensures documents are cleared and delivered in a timely fashion. The OMS serves as back up to the Ambassador’s OMS.  POCs areSandra McInturff, DCM OMS, McInturffSL@state.gov, 231-77-677-7000, ext. 7452; Samue Watson, DCM, WatsonSR@state.gov, 231-77-677-7000, ext. 7452. Please submit lobbying documents to the 360 Community Lobbying Center http://appcenter.state.sbu/CLC. (Revised 1/2017)\r\n\r\n',
        point_of_contact: null,
        website: null,
      },
      tour_of_duty: null,
      current_assignment: {
        user: 'John Doe',
        tour_of_duty: '2 YRS (2 R & R)',
        status: 'active',
        start_date: '2018-03-13T15:51:11.478084Z',
        estimated_end_date: '2020-03-13T15:51:11.478084Z',
      },
      post: {
        id: 160,
        code: 'LI6000000',
        location: {
          id: 103,
          country: 'Sierra Leone',
          code: '00A',
          city: 'Freetown',
          state: '',
        },
        tour_of_duty: '2 YRS (2 R & R)',
        cost_of_living_adjustment: 30,
        differential_rate: 30,
        danger_pay: 0,
        rest_relaxation_point: 'Paris',
        has_consumable_allowance: true,
        has_service_needs_differential: true,
        obc_id: null,
      },
      languages: [],
      bid_statistics: [
        {
          id: 6,
          bidcycle: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
          total_bids: 2,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
        },
      ],
      latest_bidcycle: {
        id: 1,
        name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
        cycle_start_date: '2017-12-13T15:51:10.980554Z',
        cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
        cycle_end_date: '2018-06-13T15:51:10.980554Z',
        active: true,
      },
    },
    {
      id: 60,
      grade: '04',
      skill: 'SECURITY (2501)',
      bureau: '(WHA) BUREAU OF WESTERN HEMISPHERIC AFFAIRS',
      organization: '(PORT AU PRIN) PORT AU PRINCE HAITI',
      classifications: [],
      representation: '[56082000] SPECIAL AGENT (Port-Au-Prince, Haiti)',
      is_highlighted: false,
      position_number: '56082000',
      title: 'SPECIAL AGENT',
      is_overseas: true,
      create_date: '2006-09-20T00:00:00Z',
      update_date: '2017-06-08T00:00:00Z',
      effective_date: '2016-10-02T00:00:00Z',
      posted_date: '2016-10-02T00:00:00Z',
      status_code: null,
      status: null,
      description: {
        id: 26,
        last_editing_user: null,
        is_editable_by_user: false,
        date_created: '2018-03-08T18:00:32.971745Z',
        date_updated: '2018-03-13T15:54:59.677323Z',
        content: '(updated August 2012) The incumbent serves as one of four Assistant Regional Security Officers (ARSO) in the Regional Security Office in a two year assignment with a rotating management portfolio that provides generous exposure to all aspects of Diplomatic Security’s overseas programs. RSO Port-au-Prince currently consists of a 6 member Marine Security Guard Detachment, 750+direct hire Mission guard force, 17 direct-hire Bodyguards, 2 FSN Investigators, 6 member Surveillance Detection Team, a 2-person Residential Security Unit, and one Office Management Specialist.  The incumbent has direct supervisory responsibilities.  The incumbent is expected to manage the Marine Security Guard program, Mission residential and physical security programs, operational and administrative oversight of the direct hire Mission guard force, and complex budget issues. The incumbent also coordinates security for large-scale Mission events and visiting high ranking USG officials and conducts personnel, terrorist, criminal, and counterintelligence investigations as required. Point of Contact is RSO Jeffrey M. Roberts.',
        point_of_contact: null,
        website: null,
      },
      tour_of_duty: null,
      current_assignment: {
        user: 'John Doe',
        tour_of_duty: '2 YRS (2 R & R)',
        status: 'active',
        start_date: '2018-03-13T15:51:13.126727Z',
        estimated_end_date: '2020-03-13T15:51:13.126727Z',
      },
      post: {
        id: 110,
        code: 'HA7000000',
        location: {
          id: 7,
          country: 'United States',
          code: '101',
          city: 'Chicago',
          state: 'IL',
        },
        tour_of_duty: '2 YRS (2 R & R)',
        cost_of_living_adjustment: 15,
        differential_rate: 20,
        danger_pay: 15,
        rest_relaxation_point: 'Miami',
        has_consumable_allowance: true,
        has_service_needs_differential: true,
        obc_id: null,
      },
      languages: [
        {
          id: 1,
          language: 'French (FR)',
          reading_proficiency: '2',
          spoken_proficiency: '2',
          representation: 'French (FR) 2/2',
        },
        {
          id: 20,
          language: 'Haitian Creole (HC)',
          reading_proficiency: '0',
          spoken_proficiency: '0',
          representation: 'Haitian Creole (HC) 0/0',
        },
      ],
      bid_statistics: [
        {
          id: 60,
          bidcycle: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
          total_bids: 0,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
        },
      ],
      latest_bidcycle: {
        id: 1,
        name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
        cycle_start_date: '2017-12-13T15:51:10.980554Z',
        cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
        cycle_end_date: '2018-06-13T15:51:10.980554Z',
        active: true,
      },
    },
  ],
};

export default resultsObject;
