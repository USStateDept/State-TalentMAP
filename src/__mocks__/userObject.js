export const bidderUserObject = {
  skill_code: ['INFORMATION MANAGEMENT (2880)', 'ADMINISTRATIVE SUPPORT (2080)'],
  user: {
    first_name: 'John',
    last_name: 'Doe',
    email: 'doej@state.gov',
    username: 'jdoe',
  },
  is_cdo: false,
  cdo: {
    email: 'shadtrachl@state.gov',
  },
  bid_statistics: [
    {
      draft: 1,
      submitted: 1,
    },
  ],
};

export const cdoUserObject = Object.assign({}, bidderUserObject, { is_cdo: true });
