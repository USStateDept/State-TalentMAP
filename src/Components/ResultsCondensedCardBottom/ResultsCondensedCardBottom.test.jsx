import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ResultsCondensedCardBottom from './ResultsCondensedCardBottom';
import resultsObject from '../../__mocks__/resultsObject';
import bidListObject from '../../__mocks__/bidListObject';
import { bidderUserObject } from '../../__mocks__/userObject';

describe('ResultsCondensedCardBottomComponent', () => {
  const type = 'default';
  const favorites = bidderUserObject.favorite_positions;
  const favoritesPV = bidderUserObject.favorite_positions_pv;
  const favoritesTandem = bidderUserObject.favorite_tandem_positions;
  const favoritesPVTandem = bidderUserObject.favorite_tandem_positions_pv;
  it('is defined', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
      />,
    );
    expect(wrapper.instance().props.type).toBe(type);
  });

  it('does not render the bid stats when showBidCount === false', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
        showBidCount={false}
      />,
    );
    expect(wrapper.find('ResultsCondensedCardStats').exists()).toBe(false);
  });

  it('renders stats if showBidCount is true', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
        showBidCount
      />,
    );
    expect(wrapper.instance().renderStats()).toBeDefined();
  });

  it('renders bid list button if showBidListButton is true', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
        showBidListButton
      />,
    );
    expect(wrapper.instance().renderBidListButton()).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when context.isClient === true', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
      />, { context: { isClient: true } },
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot with bidlist button', () => {
    const wrapper = shallow(
      <ResultsCondensedCardBottom
        position={resultsObject.results[0]}
        bidList={bidListObject.results}
        favorites={favorites}
        favoritesPV={favoritesPV}
        favoritesTandem={favoritesTandem}
        favoritesPVTandem={favoritesPVTandem}
        showBidListButton
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
