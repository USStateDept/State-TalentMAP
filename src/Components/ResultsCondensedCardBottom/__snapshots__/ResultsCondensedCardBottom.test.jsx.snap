// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsCondensedCardBottomComponent matches snapshot 1`] = `
<div
  className="condensed-card-bottom-container"
>
  <div
    className="usa-grid-full condensed-card-bottom"
  >
    <CondensedCardData
      position={
        Object {
          "bid_statistics": Array [
            Object {
              "at_skill": 0,
              "bidcycle": "Demo BidCycle 2018-03-13 15:51:10.980576+00:00",
              "id": 6,
              "in_grade": 0,
              "in_grade_at_skill": 0,
              "total_bids": 2,
            },
          ],
          "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
          "classifications": Array [],
          "create_date": "2006-09-20T00:00:00Z",
          "current_assignment": Object {
            "estimated_end_date": "2020-03-13T15:51:11.478084Z",
            "start_date": "2018-03-13T15:51:11.478084Z",
            "status": "active",
            "tour_of_duty": "2 YRS (2 R & R)",
            "user": "John Doe",
          },
          "description": Object {
            "content": "The Office Management Specialist (OMS) for the Deputy Chief of Mission (DCM) provides office management services in this Mission of six agencies, 387 American and Locally Employed Staff (LES) positions, and 279 guard force contractors.  The OMS manages Executive Office operations and positionstains the DCM’s calendar in support of overall Mission goals and objectives.  The OMS (DCM) drafts routine correspondence, memoranda and cables as appropriate and ensures documents are cleared and delivered in a timely fashion. The OMS serves as back up to the Ambassador’s OMS.  POCs areSandra McInturff, DCM OMS, McInturffSL@state.gov, 231-77-677-7000, ext. 7452; Samue Watson, DCM, WatsonSR@state.gov, 231-77-677-7000, ext. 7452. Please submit lobbying documents to the 360 Community Lobbying Center http://appcenter.state.sbu/CLC. (Revised 1/2017)

",
            "date_created": "2018-03-08T18:00:33.241370Z",
            "date_updated": "2018-03-13T15:55:00.018644Z",
            "id": 90,
            "is_editable_by_user": false,
            "last_editing_user": null,
            "point_of_contact": null,
            "website": null,
          },
          "effective_date": "2012-10-22T00:00:00Z",
          "grade": "06",
          "id": 6,
          "is_highlighted": false,
          "is_overseas": true,
          "languages": Array [],
          "latest_bidcycle": Object {
            "active": true,
            "cycle_deadline_date": "2018-04-13T15:51:10.980554Z",
            "cycle_end_date": "2018-06-13T15:51:10.980554Z",
            "cycle_start_date": "2017-12-13T15:51:10.980554Z",
            "id": 1,
            "name": "Demo BidCycle 2018-03-13 15:51:10.980576+00:00",
          },
          "organization": "(MONROVIA) MONROVIA LIBERIA",
          "position_number": "10034001",
          "post": Object {
            "code": "LI6000000",
            "cost_of_living_adjustment": 30,
            "danger_pay": 0,
            "differential_rate": 30,
            "has_consumable_allowance": true,
            "has_service_needs_differential": true,
            "id": 160,
            "location": Object {
              "city": "Freetown",
              "code": "00A",
              "country": "Sierra Leone",
              "id": 103,
              "state": "",
            },
            "obc_id": null,
            "rest_relaxation_point": "Paris",
            "tour_of_duty": "2 YRS (2 R & R)",
          },
          "posted_date": "2012-10-22T00:00:00Z",
          "representation": "[10034001] OMS (DCM) (Monrovia, Liberia)",
          "skill": "OFFICE MANAGEMENT (9017)",
          "status": null,
          "status_code": null,
          "title": "OMS (DCM)",
          "tour_of_duty": null,
          "update_date": "2017-06-08T00:00:00Z",
        }
      }
    />
    <div
      className="usa-grid-full condensed-card-buttons-section"
    >
      <Favorite
        as="div"
        className=""
        compareArray={
          Array [
            Object {
              "id": 1,
              "representation": "[00003026] OMS (COM) (Freetown, Sierra Leone)",
            },
            Object {
              "id": 4,
              "representation": "[00180000] OMS (DCM) (Addis Ababa, Ethiopia)",
            },
          ]
        }
        hasBorder={true}
        hideText={true}
        isLoading={false}
        onToggle={[Function]}
        refKey={6}
        useButtonClass={true}
        useButtonClassSecondary={true}
        useLongText={false}
        useSpinnerWhite={false}
      />
      <BidListButton
        className="tm-button"
        compareArray={
          Array [
            Object {
              "approved_date": "2017-12-20",
              "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
              "closed_date": "2017-12-20",
              "create_date": "2017-12-20",
              "declined_date": "2017-12-20",
              "draft_date": "2017-12-20",
              "handshake_accepted_date": "2017-12-20",
              "handshake_declined_date": "2017-12-20",
              "handshake_offered_date": "2017-12-20",
              "id": 3,
              "in_panel_date": "2017-12-20",
              "is_priority": false,
              "position": Object {
                "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
                "create_date": "2015-08-05",
                "grade": "03",
                "id": 226,
                "position_number": "10035561",
                "post": Object {
                  "id": 199,
                  "location": Object {
                    "city": "Freetown",
                    "code": "00A",
                    "country": "Sierra Leone",
                    "id": 103,
                    "state": "",
                  },
                },
                "skill": "POLITICAL AND ECONOMIC AFFAIRS (6050)",
                "title": "POLITICAL/ECONOMIC OFFICER",
                "update_date": "2017-06-08",
              },
              "reviewer": Object {
                "email": "woodwardw@state.gov",
                "first_name": "Wendy",
                "is_cdo": false,
                "last_name": "Woodward",
                "phone_number": "555-555-5555",
                "username": "woodwardw",
              },
              "scheduled_panel_date": "2017-12-20",
              "status": "approved",
              "submitted_date": "2017-12-20",
              "update_date": "2017-12-20",
              "user": "rehmant",
            },
            Object {
              "approved_date": "2017-12-20",
              "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
              "closed_date": "2017-12-20",
              "create_date": "2017-12-20",
              "declined_date": "2017-12-20",
              "draft_date": "2017-12-20",
              "handshake_accepted_date": "2017-12-20",
              "handshake_declined_date": "2017-12-20",
              "handshake_offered_date": "2017-12-20",
              "id": 1,
              "in_panel_date": "2017-12-20",
              "is_priority": true,
              "position": Object {
                "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
                "create_date": "2006-09-20",
                "grade": "05",
                "id": 1,
                "position_number": "00003026",
                "post": Object {
                  "id": 235,
                  "location": Object {
                    "city": "Freetown",
                    "code": "00A",
                    "country": "Sierra Leone",
                    "id": 103,
                    "state": "",
                  },
                },
                "skill": "OFFICE MANAGEMENT (9017)",
                "title": "OMS (COM)",
                "update_date": "2017-06-08",
              },
              "reviewer": Object {
                "email": "woodwardw@state.gov",
                "first_name": "Wendy",
                "is_cdo": false,
                "last_name": "Woodward",
                "phone_number": "555-555-5555",
                "username": "woodwardw",
              },
              "scheduled_panel_date": "2017-12-20",
              "status": "closed",
              "submitted_date": "2017-12-20",
              "update_date": "2017-12-20",
              "user": "rehmant",
            },
            Object {
              "approved_date": "2017-12-20",
              "bidcycle": "Demo BidCycle 2017-10-31 16:00:51.254762",
              "closed_date": "2017-12-20",
              "create_date": "2017-12-20",
              "declined_date": "2017-12-20",
              "draft_date": "2017-12-20",
              "handshake_accepted_date": "2017-12-20",
              "handshake_declined_date": "2017-12-20",
              "handshake_offered_date": "2017-12-20",
              "id": 2,
              "in_panel_date": "2017-12-20",
              "is_priority": false,
              "position": Object {
                "bureau": "(AF) BUREAU OF AFRICAN AFFAIRS",
                "create_date": "2006-09-20",
                "grade": "04",
                "id": 84,
                "position_number": "56001600",
                "post": Object {
                  "id": 195,
                  "location": Object {
                    "city": "Freetown",
                    "code": "00A",
                    "country": "Sierra Leone",
                    "id": 103,
                    "state": "",
                  },
                },
                "skill": "SECURITY (2501)",
                "title": "SPECIAL AGENT",
                "update_date": "2017-06-08",
              },
              "reviewer": Object {
                "email": "woodwardw@state.gov",
                "first_name": "Wendy",
                "is_cdo": false,
                "last_name": "Woodward",
                "phone_number": "555-555-5555",
                "username": "woodwardw",
              },
              "scheduled_panel_date": "2017-12-20",
              "status": "closed",
              "submitted_date": "2017-12-20",
              "update_date": "2017-12-20",
              "user": "rehmant",
            },
          ]
        }
        id={6}
        isLoading={false}
        toggleBidPosition={[Function]}
      />
    </div>
  </div>
</div>
`;
